commitizen-tools/commitizen
# Add `version_type` option to make version compatibly with semver

## Description

I decided to use it in a rust project but cargo crashed when trying to use the prerelease versions.

I have added an optional setting that solves my problem for now.


### `version_type`

Choose version type

* `pep440` - default version provider.
    - prerelease - `1.0.1a0`
    - devrelease - `1.0.1dev0`
    - dev and pre - `1.0.1a0.dev0`
* `semver` - semver compatibly provider. Added "-" delimiter
    - prerelease - `1.0.1-a0`
    - devrelease - `1.0.1-dev0`
    - dev and pre - `1.0.1-a0-dev0`


TODO: renamed pre to long name, like alpha, beta, ..etc via additional option `semver_long`

```toml
[tool.commitizen]
version_type = "semver"
```

## Checklist


## Expected behavior



## Steps to Test This Pull Request
```shell
$ cz bump --check-consistency --no-verify --prerelease alpha --version-type=semver --dry-run
bump: version 0.1.0-a2 → 0.1.0-a3
tag to create: v0.1.0-a3

```


## Additional context

#150
